test/hpke_test.c

```diff
@@ -0,0 +1,1801 @@
+    if (!TEST_true(OSSL_HPKE_encap(ctx, enc, &enclen, pub, publen, NULL, 0)))
+        goto end;
+    /* second encap fail */
+    if (!TEST_false(OSSL_HPKE_encap(ctx, enc, &enclen, pub, publen, NULL, 0)))
+        goto end;
+    plainlen = 0;
+    /* should fail for no plaintext */
+    if (!TEST_false(OSSL_HPKE_seal(ctx, cipher, &cipherlen, NULL, 0,
+                                   plain, plainlen)))
+        goto end;
+    /* the sequence ought not have been incremented, so good to start over */
+    plainlen = sizeof(plain);
+    /* seq wrap around test */
+    if (!TEST_true(OSSL_HPKE_CTX_set_seq(ctx, -1)))
+        goto end;
+    if (!TEST_false(OSSL_HPKE_seal(ctx, cipher, &cipherlen, NULL, 0,
+                                   plain, plainlen)))
+        goto end;
+    /* reset seq */
+    if (!TEST_true(OSSL_HPKE_CTX_set_seq(ctx, 0)))
+        goto end;
+    /* working seal */
+    if (!TEST_true(OSSL_HPKE_seal(ctx, cipher, &cipherlen, NULL, 0,
+                                  plain, plainlen)))
+        goto end;
+
+    /* receiver side */
+    /* decap fail with psk mode but no psk set */
+    if (!TEST_ptr(rctx = OSSL_HPKE_CTX_new(OSSL_HPKE_MODE_PSK, hpke_suite,
+                                           testctx, NULL)))
+        goto end;
+    if (!TEST_false(OSSL_HPKE_decap(rctx, enc, enclen, privp, NULL, 0)))
+        goto end;
+    /* done with PSK mode */
+    OSSL_HPKE_CTX_free(rctx);
+
+    /* back good calls for base mode  */
+    if (!TEST_ptr(rctx = OSSL_HPKE_CTX_new(hpke_mode, hpke_suite,
+                                           testctx, NULL)))
+        goto end;
+    /* open before decap */
+    if (!TEST_false(OSSL_HPKE_open(rctx, clear, &clearlen, NULL, 0,
+                                   cipher, cipherlen)))
+        goto end;
+    /* decap with info too long */
+    if (!TEST_false(OSSL_HPKE_decap(rctx, enc, enclen, privp, info, -1)))
+        goto end;
+    /* good decap */
+    if (!TEST_true(OSSL_HPKE_decap(rctx, enc, enclen, privp, NULL, 0)))
+        goto end;
+    /* second decap fail */
+    if (!TEST_false(OSSL_HPKE_decap(rctx, enc, enclen, privp, NULL, 0)))
+        goto end;
+    /* no space for recovered clear */
+    clearlen = 0;
+    if (!TEST_false(OSSL_HPKE_open(rctx, clear, &clearlen, NULL, 0,
+                                   cipher, cipherlen)))
+        goto end;
+    clearlen = OSSL_HPKE_TSTSIZE;
+    /* seq wrap around test */
+    if (!TEST_true(OSSL_HPKE_CTX_set_seq(rctx, -1)))
+        goto end;
+    if (!TEST_false(OSSL_HPKE_open(rctx, clear, &clearlen, NULL, 0,
+                                   cipher, cipherlen)))
+        goto end;
+    if (!TEST_true(OSSL_HPKE_CTX_set_seq(rctx, 0)))
+        goto end;
+    if (!TEST_true(OSSL_HPKE_open(rctx, clear, &clearlen, NULL, 0,
+                                  cipher, cipherlen)))
+        goto end;
+    if (!TEST_true(TEST_mem_eq(plain, plainlen, clear, clearlen)))
+        goto end;
+
+    OSSL_HPKE_CTX_free(ctx);
```

Comment:
duplicated code .
just set a `result = 1 `before the end: label 
and so you could `return result` ate the end of the function.

Reply:
sure